src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -224,9 +224,15 @@ public static string ObjectTemplate(IHtmlHelper htmlHelper)
 
             var serviceProvider = htmlHelper.ViewContext.HttpContext.RequestServices;
             var viewEngine = serviceProvider.GetRequiredService<ICompositeViewEngine>();
-            var properties = modelMetadata.Properties.Where(metadata => ShouldShow(metadata, templateInfo));
-            foreach (var propertyMetadata in properties)
+
+            foreach (var propertyMetadata in modelExplorer.Metadata.Properties)
             {
+                var propertyExplorer = modelExplorer.GetExplorerForProperty(propertyMetadata.PropertyName);
```

Comment:
worth adding an `IEnumerable<ModelExplorer> Properties { get; }` to `ModelExplorer`?  done here, in the editor template, anywhere else?